src/middlewared/middlewared/service.py

```diff
@@ -871,6 +872,32 @@ def ping(self):
         """
         return 'pong'
 
+    def _ping_host(self, host, timeout):
+        if osc.IS_LINUX:
+            process = run(['ping', '-w', f'{timeout}', host])
+        else:
+            process = run(['ping', '-t', f'{timeout}', host])
+
+        if process.returncode != 0:
+            return False
+        else:
+            return True
+
+    @accepts(
+        Str('hostname'),
+        Int('timeout', validators=[Range(min=1, max=60)], default=10),  # seconds
+    )
```

Comment:
Lets make extend this API interface to be more future-proof, please.

Something like:

```
Dict(
    'options',
    Str('type', enum=['ICMP', 'ICMPV4', 'ICMPV6'], default='ICMP'),
    Str('hostname', required=True),
    Int('timeout', validators=[Range(min=1, max=60)], default=10),
)
```

We will probably have to resolve the hostname first and give preference to IPv4.

This way we could extend it in the future to include more protocols without breaking the API (backward compatible)